### PR TITLE
chore(CRT-281): create KubeFedCluster CRD programmatically

### PR DESCRIPTION
**Pull request by MatousJobanek**
We need to create KubeFedCluster CRD programmatically because we cannot install it via CSV - for more information see this issue: operator-framework/operator-lifecycle-manager#1060

This PR:
* removes `core_v1beta1_kubefedcluster_crd.yaml` file and updates CSVs
* calls a function `EnsureKubeFedClusterCrd` from the common repo that checks if the CRD exists, if it does not exist, then it creates it

This PR depends on:
https://github.com/codeready-toolchain/toolchain-common/pull/44
https://github.com/codeready-toolchain/api/pull/43

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -44,6 +46,10 @@ github.com/MakeNowJust/heredoc v0.0.0-20171113091838-e9091a26100e/go.mod h1:64YH
 github.com/Masterminds/goutils v1.1.0/go.mod h1:8cTjp+g8YejhMuvIA5y2vz3BpJxksy863GQaJW2MFNU=
 github.com/Masterminds/semver v1.4.2/go.mod h1:MB6lktGJrhw8PrUyiEoblNEGEQ+RzHPF078ddwwvV3Y=
 github.com/Masterminds/sprig v0.0.0-20190301161902-9f8fceff796f/go.mod h1:y6hNFY5UBTIWBxnzTeuNhlNS5hqE0NB0E6fgfo2Br3o=
+github.com/MatousJobanek/api v0.0.0-20191008135136-1338bbea618f h1:H06CSIL/t13A+JFl7j+sK6LC2dIPgviS+9EeWvmXclo=
+github.com/MatousJobanek/api v0.0.0-20191008135136-1338bbea618f/go.mod h1:/Is7p+crldTqYG0f8LQL7OqmdQRnMJ7u//N82tQdf28=
+github.com/MatousJobanek/toolchain-common v0.0.0-20191008140617-66eaba2d7197 h1:J9D87vF+2taLCPAVILnadDnopmkDn7fRUjHROxzvdvg=
+github.com/MatousJobanek/toolchain-common v0.0.0-20191008140617-66eaba2d7197/go.mod h1:nCOcq3lAyVk9XjWei07rt/MEqvQjrMHnOr8mxspI3mk=
```
Comment (alexeykazakov) [inline go.sum]: Same here.

File: pkg/controller/kubefedcluster.go [hunk 2]
```diff
@@ -1,15 +1,19 @@
...
 	"github.com/operator-framework/operator-sdk/pkg/k8sutil"
 	"k8s.io/klog"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
 	"sigs.k8s.io/kubefed/pkg/controller/kubefedcluster"
 	"sigs.k8s.io/kubefed/pkg/controller/util"
 )
 
-func StartKubeFedClusterControllers(mgr manager.Manager, stopChan <-chan struct{}) error {
+func StartKubeFedClusterControllers(log logr.Logger, mgr manager.Manager, stopChan <-chan struct{}) error {
+	cluster.EnsureKubeFedClusterCrd(log, mgr, stopChan)
```
[the remark this answers is not in the document]
Comment (MatousJobanek) [reply pkg/controller/kubefedcluster.go]: done in bc13116

File: go.mod [hunk 3]
```diff
@@ -36,6 +38,8 @@ require (
 	sigs.k8s.io/kubefed v0.1.0-rc2
 )
 
+replace github.com/codeready-toolchain/toolchain-common => github.com/MatousJobanek/toolchain-common v0.0.0-20191009091227-b631ea4cbdc3
```
Comment (alexeykazakov) [inline go.mod]: !

File: go.mod [hunk 4]
```diff
@@ -36,6 +38,10 @@ require (
 	sigs.k8s.io/kubefed v0.1.0-rc2
 )
 
+replace github.com/codeready-toolchain/api => github.com/MatousJobanek/api v0.0.0-20191008135136-1338bbea618f
+
+replace github.com/codeready-toolchain/toolchain-common => github.com/MatousJobanek/toolchain-common v0.0.0-20191008140617-66eaba2d7197
```
[the remark this answers is not in the document]
Comment (MatousJobanek) [reply go.mod]: Thanks for the reminder :-) 45efee4

File: go.sum (hunk 1 above)
Comment (MatousJobanek) [reply go.sum]: Thanks for the reminder :-) 45efee4

File: go.mod (hunk 3 above)
Comment (MatousJobanek) [reply go.mod]: see: 45efee4 
* removed replace
* updated common
* tidied

**Pull request merged**